### ISSUE TITLE
Non-ASCII UTF-8 in SDL2 window title leads to mojibake in Plasmashell taskbar, but Firefox and other applications don't have this problem

### ISSUE THREAD

**Issue by ctrlcctrlv** [aclysma/skulpin#93]
There's something wrong with the way Skulpin is wrapping raw window handles.

I verified this by trying to add the same type of text to the SDL2 demo (while also modifying the SDL2 demo to be closer to Skulpin):

![image](https://user-images.githubusercontent.com/838783/113849459-84c71f80-9767-11eb-9d79-1093a927eded.png)

```diff
diff --git a/examples/demo.rs b/examples/demo.rs
index 7521fa9b..838f4b5c 100644
--- a/examples/demo.rs
+++ b/examples/demo.rs
@@ -10,9 +10,11 @@ pub fn main() -> Result<(), String> {
     let video_subsystem = sdl_context.video()?;
 
     let window = video_subsystem
-        .window("rust-sdl2 demo: Video", 800, 600)
+        .window("文字化けじゃない", 800, 600)
         .position_centered()
-        .opengl()
+        .allow_highdpi()
+        .vulkan()
+        .resizable()
         .build()
         .map_err(|e| e.to_string())?;
```

Modifying the Skulpin demo, however, does result in mojibake in the taskbar.

```diff --git a/examples/hello_skulpin_sdl2.rs b/examples/hello_skulpin_sdl2.rs
index 913a384..ed88235 100644
--- a/examples/hello_skulpin_sdl2.rs
+++ b/examples/hello_skulpin_sdl2.rs
@@ -34,7 +34,7 @@ fn main() {
     };
 
     let window = video_subsystem
-        .window("Skulpin", logical_size.width, logical_size.height)
+        .window("Skulpin — \u{6587}\u{5B57}\u{5316}\u{3051}", logical_size.width, logical_size.height)
         .position_centered()
         .allow_highdpi()
         .resizable()

```

![image](https://user-images.githubusercontent.com/838783/113849812-d40d5000-9767-11eb-8c7f-fcf4a1d522f2.png)

Note taskbar, **Skulpin — 文字化け** ⇒ **Skulpin â æå  åã**.

I need help debugging this further, don't even know where to start. Plasmashell is either the most popular, or second most popular, taskbar for Linux as it's part of the KDE project.

![image](https://user-images.githubusercontent.com/838783/113850421-74637480-9768-11eb-9327-11067be2802d.png)

What is Firefox doing right that Skulpin is doing wrong? :thinking: 

Note: Skulpin displays text correctly if Winit in use.

![image](https://user-images.githubusercontent.com/838783/113851668-bb9e3500-9769-11eb-9d28-9a60d1aca49b.png)

```diff
diff --git a/examples/hello_skulpin_winit.rs b/examples/hello_skulpin_winit.rs
index 87f3f2e..fbfc885 100644
--- a/examples/hello_skulpin_winit.rs
+++ b/examples/hello_skulpin_winit.rs
@@ -18,7 +18,7 @@ fn main() {
     // Set up the coordinate system to be fixed at 900x600, and use this as the default window size
     // This means the drawing code can be written as though the window is always 900x600. The
     // output will be automatically scaled so that it's always visible.
-    let logical_size = winit::dpi::LogicalSize::new(900.0, 600.0);
+    let logical_size = winit::dpi::PhysicalSize::new(900.0, 600.0);
     let visible_range = skulpin::skia_safe::Rect {
         left: 0.0,
         right: logical_size.width as f32,
@@ -29,7 +29,7 @@ fn main() {
 
     // Create a single window
     let window = winit::window::WindowBuilder::new()
-        .with_title("Skulpin")
+        .with_title("Skulpin — 文字化けじゃない")
         .with_inner_size(logical_size)
         .build(&event_loop)
         .expect("Failed to create window");
```

**Comment by aclysma** [comment]
I don't use desktop linux and won't be able to help with this. Skulpin does not call into SDL2 APIs. If the SDL2 example works and the skulpin examples don't work, I would verify that you're using the same version of the sdl2 and sdl2-sys crates.

**Comment by ctrlcctrlv** [comment]
This is a bug in your own example, though? If you're not going to maintain the examples…

**Comment by ctrlcctrlv** [comment]
Hello, good news; although you prematurely closed this, I tracked the issue through Rust-SDL2/rust-sdl2 all the way to libsdl-org/SDL. I patched SDL itself, see libsdl-org/SDL#4290. I've confirmed that this issue fixes Skulpin as well. So libsdl-org/SDL#4290 fixes this bug, Rust-SDL2/rust-sdl2#1087, and libsdl-org/SDL#4288, which all turn out to be caused by libsdl-org/SDL#4288.

![image](https://user-images.githubusercontent.com/838783/113894072-a7702d00-9795-11eb-937d-855ca25f4264.png)
